contents/blog/decouple-deployment-from-release.md

```diff
@@ -0,0 +1,114 @@
+
+A best practice for improving this is decoupling deployment and release. What does this mean, though? This post goes over the differences between the two, why you should decouple them, examples, and how to get started.
+
+## What are the differences between deployment and release?
+
+The process of delivering code starts with writing and testing code locally. After completing a piece of code, it often goes through reviews and more tests before integrating with the existing codebase. 
+
+Once integrated, the new code deploys. Deployment means getting code from a local machine or repository into production. Your app or build containing the new code becomes available on production infrastructure.
+
+A release is when new code is put into use or made available to users. By default, as soon as code is deployed, it is automatically released. This is because there is nothing preventing users or systems from accessing the new code available on production infrastructure.
+
+This means teams often hold deployments until they are confident the code is ready to release.
+
+![Delivery process](../images/blog/decouple-deployment-from-release/delivery.png)
+
+## Why should you separate deployment and release?
+
+The coupling of deployment and release can cause two main issues:
+
+1. New code doesn't work as expected. For example, a value saving locally but not in production due to an unexpected interaction with the production database.
+
+2. It breaks other features or the entire app. For example, values created by new features corrupt other data unexpectedly and crashes the app in production.
+
+By preventing new code from automatically releasing after a deployment, you lower the risk of these issues. This is known as **decoupling deployment from release**.
+
+Doing this enables [testing code in production](/blog/testing-in-production). This process helps you discover issues arising from production infrastructure and the new code's integration with the real release.
+
+On top of lowering the risk of issues, decoupling can increase developer productivity and app reliability. The specific benefits include:
+
+- Improves developer confidence knowing they can test in production and that there is an easy way to roll back changes without affecting users.
+
+- Increase deployment frequency by removing the need to wait for a formal release. 
+
+- Developers can follow [trunk-based development](https://trunkbaseddevelopment.com/) and merge to `main` more often instead of waiting on branches. This increases shipping velocity and reduces merge conflicts.
+
+- Developers can pass off releases to product managers and marketers once they deploy. They can coordinate elaborate launches and stakeholders without holding up development.
+
+- Less downtime from fewer broken releases. Easier rollbacks when an issue does occur because infrastructure (like [feature flags](/feature-flags)) is already in place.
```

Comment:
I know we're trying to link to feature flags, but it feels a bit unnaturally verbose
```suggestion
- Less downtime from broken releases, and easier rollbacks when an issue does occur.
```